sql-bricks.js

```diff
@@ -3,12 +3,81 @@
 
   var is_common_js = typeof exports != 'undefined';
   var default_opts = { placeholder: '$%d' };
-  
-  var _;
-  if (is_common_js)
-    _ = require('underscore');
-  else
-    _ = window._;
+
+  function toArray(obj) {
+    return Object.keys(obj).map(function(key) {
+      return obj[key];
+    });
+  }
+
+  function extend(obj) {
+    var other_objs = arguments;
+    delete other_objs['0'];
+
+    Object.keys(other_objs).forEach(function(arg_num) {
+      var other_obj = other_objs[arg_num];
+      Object.keys(other_obj).forEach(function(key) {
+        obj[key] = other_obj[key];
+      });
+    });
+    return obj;
+  }
+
+  function cloneObj(obj) {
+    if (types.isArray(obj))
+      return [].concat(arr);
+    else
+      return extend({}, obj);
+  }
+
+  var types = {
+    isObject: function isObject(val) {
+      return typeof val == 'object';
+    },
+    isArray: function isArray(val) {
+      return val instanceof Array;
+    },
+    isUndefined: function isUndefined(val) {
+      return typeof val == 'undefined';
+    },
+    isNull: function isNull(val) {
+      return val === null;
+    },
+    isNumber: function isNumber(val) {
+      return typeof val == 'number';
+    },
+    isString: function isString(val) {
+      return typeof val == 'string';
+    },
+    isBoolean: function isBoolean(val) {
+      return typeof val == 'boolean';
+    },
+    isDate: function isDate(val) {
+      return val instanceof Date;
+    }
+  };
+
+  function findIndex(arr, fn) {
+    var arr_index = -1;
+    arr.forEach(function(val, index) {
+      if (fn(val)) {
+        arr_index = index;
+        break;

```

Comment:
@schuttsm: This failed in Travis's [Node 6](https://travis-ci.org/CSNW/sql-bricks/jobs/601886566?utm_medium=notification&utm_source=github_status) test run:

```
/home/travis/build/CSNW/sql-bricks/sql-bricks.js:65
        break;
        ^^^^^

SyntaxError: Illegal break statement
```

Not that I think we still need to be supporting Node 6, but I don't think `break` is legal inside a `forEach()`. It might be easiest to just do a regular `for` loop here...